src/Grids/extruded.jl

```diff
@@ -0,0 +1,120 @@
+#####
+##### Hybrid mesh
+#####
+
+abstract type HypsographyAdaption end
+
+"""
+    Flat()
+
+No surface hypsography.
+"""
+struct Flat <: HypsographyAdaption end
+
+abstract type AbstractExtrudedFiniteDifferenceGrid <: AbstractGrid end
+
+"""
+    ExtrudedFiniteDifferenceGrid(
+        horizontal_space::AbstractSpace,
+        vertical_space::FiniteDifferenceSpace,
+        hypsography::HypsographyAdaption = Flat(),
+    )
+
+Construct an `ExtrudedFiniteDifferenceGrid` from the horizontal and vertical spaces.
+"""
+mutable struct ExtrudedFiniteDifferenceGrid{
+    H <: AbstractGrid,
+    V <: FiniteDifferenceGrid,
+    A <: HypsographyAdaption,
+    GG <: Geometry.AbstractGlobalGeometry,
+    LG,
+} <: AbstractExtrudedFiniteDifferenceGrid
+    horizontal_grid::H
+    vertical_grid::V
+    hypsography::A
+    global_geometry::GG
+    center_local_geometry::LG
+    face_local_geometry::LG
+end
+
+@memoize WeakValueDict function ExtrudedFiniteDifferenceGrid(
```

Comment:
Can we document (or provide a link to docs to show) what `@memoise WeakValueDict` is doing here? or somewhere?